File: Excel/535 Challenge.py
import pandas as pd
import numpy as np

path = "535 Generate Number Grid.xlsx"
height = pd.read_excel(path, usecols = "B", nrows = 1, header = None).values[0][0]
width = pd.read_excel(path, usecols = "B", nrows = 1, skiprows = 1, header = None).values[0][0]
test = pd.read_excel(path, usecols = "D:K", skiprows = 1, header = None).values

matrix = np.zeros((height, width), dtype=int)

for i in range(1, height + 1):
    matrix[i - 1, :] = np.roll(np.arange(1, width + 1), i - 1)

print(np.array_equal(matrix, test)) # True

File: Excel/500-599/535/535 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/535 Generate Number Grid.xlsx"
height = read_excel(path, range = "B1", col_names = F) %>% pull()
width = read_excel(path, range = "B2", col_names = F) %>% pull()

test = read_excel(path, range = "D2:K7", col_names = F) %>% as.matrix()

matrix = matrix(0, nrow = height, ncol = width)
for (i in 1:height) {
  matrix[i, ] <- c(tail(1:width, i - 1), head(1:width, width - (i - 1)))
}

all.equal(matrix, test, check.attributes = F)
# [1] TRUE